spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -573,7 +574,8 @@ public static void importSparkPartitions(
     int numShufflePartitions = spark.sessionState().conf().numShufflePartitions();
     MetricsConfig metricsConfig = MetricsConfig.fromProperties(targetTable.properties());
     String nameMappingString = targetTable.properties().get(TableProperties.DEFAULT_NAME_MAPPING);
-    NameMapping nameMapping = nameMappingString != null ? NameMappingParser.fromJson(nameMappingString) : null;
+    NameMapping nameMapping = nameMappingString != null ?
+        NameMappingParser.fromJson(nameMappingString) : MappingUtil.create(targetTable.schema());
```

Comment:
This change needs to be reverted. The import process must create and store a mapping on the table. We should not use an ephemeral mapping that is immediately discarded.

Reply:
Yeah, somehow I missed this one but did it for the method of unpartitioned tables. Thanks for pointing this out. I've reverted the change.

Reply:
That explains it. I thought that you were going to, so I was surprised.